packages/common-all/src/types/configs/commands/copyNoteLink.ts

```diff
@@ -0,0 +1,16 @@
+/** "line" uses line numbers (`L23`), "block" inserts block anchors (`^xf1g...`). If undefined prompt user to select one. */
+export type NonNoteFileLinkAnchorType = "line" | "block" | undefined;
+
+/**
+ * Namespace for configuring {@link CopyNoteLinkCommand}
+ */
+export type CopyNoteLinkConfig = {
+  nonNoteFile?: {
+    anchorType?: NonNoteFileLinkAnchorType;
+  };
+};
+
+export function genDefaultCopyNoteLinkConfig(): CopyNoteLinkConfig {
+  // don't set a default for `nonNoteFiles`, we want to prompt the user whether they want lines or block anchors
```

Comment:
rationale for prompting users vs sticking with the behavior we have for note files? (which is adding an anchor block)

I think there's an argument to be made whether that is the right default but unless there's good reason (and there might be - I can see users not wanting to accidentally add an anchor to some external file), we should have the same default for note files as well as non note files

In which case, we might want to revisit or default for copy note link and set the default there to be a prompt as well (followup task)

Reply:
I added the option because that's what we previously discussed to do :sweat_smile: 

I think that makes sense. I think with notes it might be better to use the read-only status introduced in RFC 42 later so we generate line numbers for read-only vaults and block anchors for writeable vaults. But we can iterate on the UX of that later.

Reply:
yep, i remember making that proposal 😅
now seeing it in action, I realize that I was wrong